exercises/nodejs/src/game.js

```diff
@@ -1,172 +1,198 @@
+    );
+    console.log("The category is " + currentCategory());
+    askQuestion();
+  };
+
+
+  this.actionAfterRoll = function (roll) {
+    console.log(players[currentPlayer] + " is the current player");
+    console.log("They have rolled a " + roll);
+    if (this.playerIsInPenaltyBox() && this.playerGetsOutOfPenaltyBox(roll)) {
+      isGettingOutOfPenaltyBox = true;
+      console.log(
+        players[currentPlayer] + " is getting out of the penalty box"
+      );
+      this.playerTakesTurn(roll);
+    } else if (this.playerIsInPenaltyBox() & !this.playerGetsOutOfPenaltyBox(roll)) {
+      console.log( players[currentPlayer] + " is not getting out of the penalty box" );
+      isGettingOutOfPenaltyBox = false;
+    } else {
+      this.playerTakesTurn(roll);
+    }
+  };
 
-  this.wrongAnswer = function () {
-    console.log('Question was incorrectly answered')
-    console.log(players[currentPlayer] + ' was sent to the penalty box')
-    inPenaltyBox[currentPlayer] = true
 
-    currentPlayer += 1
-    if (currentPlayer == players.length) { currentPlayer = 0 }
-    return true
-  }
-}
+  this.moveToNextPlayer = function () {
+    currentPlayer += 1;
+    if (currentPlayer == players.length) {
+      currentPlayer = 0;
+    }
+  };
 
+  this.answerWasCorrect = function () {
+    if (inPenaltyBox[currentPlayer]) {
+      if (isGettingOutOfPenaltyBox) {
+        console.log("Answer was correct!!!!");
+        playersScore[currentPlayer] += 1;
+        console.log( players[currentPlayer] + " now has " +  playersScore[currentPlayer] + " Gold Coins." );
+        var winner = didPlayerWin();
+        this.moveToNextPlayer();
+        return winner;
+      } else {
+        this.moveToNextPlayer();
+        return true;
+      }
+    } else {
+      console.log("Answer was correct!!!!");
+      playersScore[currentPlayer] += 1;
+      console.log(players[currentPlayer] + " now has " + playersScore[currentPlayer] +" Gold Coins." );
+      var winner = didPlayerWin(); //if true, they don’t have 6 coins
+      //if false, they do have six coins?
+      this.moveToNextPlayer();
+      //turns - if the last player has been go back to player 0
+      return winner;
+    }
+  };
+  this.answerWasWrongAndCheckThereIsNoWinner = function () {
+    console.log("Question was incorrectly answered");
+    console.log(players[currentPlayer] + " was sent to the penalty box");
+    inPenaltyBox[currentPlayer] = true;
+    //they are in the penalty box, but can keep playing to get out of this
+    currentPlayer += 1;
+    if (currentPlayer == players.length) {
+      currentPlayer = 0;
+    }
+    return true;
+    //checks turn - if this player is the last player in the set of of people, go back to player 0
+  };
+};
 const gameRunner = (i) => {
-  var notAWinner = false
-
-  var game = new Game()
-
-  game.add('Chet')
-  game.add('Pat')
-  game.add('Sue')
-
-  const random = generator.create(i)
-
+  var notAWinner = false;
```

Comment:
notAWinner is a confusing variable name as the opposite as this stays false until there is a winner, should have the opposite name or opposite boolean effect.